components/speedreader/rust/lib/src/readability/src/scorer.rs

```diff
@@ -744,7 +801,30 @@ pub fn clean<S: ::std::hash::BuildHasher>(
                 | local_name!("header")
                 | local_name!("footer")
                 | local_name!("embed")
+                | local_name!("textarea")
+                | local_name!("input")
+                | local_name!("select")
+                | local_name!("button")
                 | local_name!("aside") => true,
+                local_name!("h1") | local_name!("h2") => {
+                    // Delete remaining headings that may be duplicates of the title.
+                    let mut heading = String::new();
+                    dom::extract_text(&handle, &mut heading, true);
+                    if heading.len() == 0 {
+                        return true;
+                    }
+                    if title_tokens.len() == 0 {
+                        return false;
+                    }
```

Comment:
`heading.is_empty()`, `title_tokens.is_empty()` - I believe these would have been caught by a `cargo clippy` run

Reply:
fixed, and i also made some other `cargo clippy` fixes introduced in this patchset. i will do a separate clippy pass for the others.